cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/wizard/common/model/SelectFileContentProvider.java

```diff
@@ -461,9 +462,14 @@ public Object getProvidedObject(String path) {
      * @return <code>true</code> if the last element of the path contains a dot<br>
      *         <code>false</code>, otherwise
      */
-    private boolean targetIsFile(IPath path) {
-
-        return path.lastSegment().contains(".");
+    private boolean targetIsFile(IPath path, List<String> paths) {
+        int index = paths.indexOf(path.toString());
+        if (index + 1 <= paths.size() - 1) {
```

Comment:
Can you explain me the logic behind that `if`?

Wouldn't be the same with the following?:
`if (index < path.size())`

Reply:
I think the only correct subsitution would be if (index < paths.size() -1).
Your's @jdiazgon would be failing.

Reply:
Ooh yeah, you are right